lms/djangoapps/grades/new/course_grade.py

```diff
@@ -328,29 +328,32 @@ def update(self, course):
         course_structure = get_course_blocks(self.student, course.location)
         self._compute_and_update_grade(course, course_structure)
 
-    def _compute_and_update_grade(self, course, course_structure, read_only=False):
-        """
-        Freshly computes and updates the grade for the student and course.
-
-        If read_only is True, doesn't save any updates to the grades.
-        """
-        course_grade = CourseGrade(self.student, course, course_structure)
-        course_grade.compute_and_update(read_only)
-        return course_grade
-
-    def _get_saved_grade(self, course, course_structure):  # pylint: disable=unused-argument
+    def get_persisted(self, course, course_structure=None):
         """
         Returns the saved grade for the given course and student.
         """
         if not PersistentGradesEnabledFlag.feature_enabled(course.id):
             return None
 
+        if course_structure is None:
+            course_structure = get_course_blocks(self.student, course.location)
```

Comment:
This isn't acceptable from a performance perspective. - We only want to get the data from the database. - We don't want to recompute the value at all when displaying grades on the dashboard.